10-walkthrough-5.Rmd

```diff
@@ -8,74 +8,229 @@ knitr::opts_chunk$set(echo = TRUE, message = F, warning = F, cache = F)
+
+```{r, echo = F}
 naep17 <-
-    readxl::read_xlsx(here::here("data", "naep", "2017_Results_Appendix_Math_State.xlsx"), sheet = 5, skip = 8)
+    readxl::read_xlsx(
+        here::here("data", "naep", "NDECoreExcel_Mathematics, Grade 8, Race ethnicity used to repor_20190825230831.xlsx"), 
+        skip = 8)
+
+str(naep17) # notice the proficiency rates are imported as characters
 ```
 
 ### Cleaning
 
-Once imported, the dataset contains a column with different states/jurisdiction, columns with the various years and the proficiency rates for each one, and miscellaenous columns denoting statistic significance. There are also rows at the bottom with additional information on the dataset. For the purpose of our walkthrough, we want the dataset to be as clean and simple as possible. We will remove the columns with significance status, the rows with additional information, and transform the proficiency rates to numeric.
+The dataset contains a column with year, states/jurisdiction, race/ethnicity, and the proficiency rates for each one, and miscellaneous columns denoting statistic significance. There are also rows at the bottom with additional information on the dataset. For the purpose of our walkthrough, we want the dataset to be as clean and simple as possible.
+
+- `janitor` is an excellent package with many handy functions, including `clean_names()` that transforms column names to R-friendly formats (such as removing spaces, making every word lowercase).
+- `dplyr::slice()` selects rows by positions. Because we want to remove rows, the row numbers are negative. Another way of writing this line is `dplyr::slice(1:318)`. *Caution:* all rows selected (either to keep or remove) must have the same sign (either all positive or all negative).
+
+```{r, warning = FALSE}
+naep17_tidy <-
+    naep17 %>%
+    janitor::clean_names() %>% 
+    dplyr::slice(-319:-342) # remove rows containing additional information
+    
+colnames(naep17_tidy)
+```
+
+The dataset denotes statistical significance in column `at_or_above_proficient` with an asterisk next to the percentage (e.g., 44*). If we change this column to numeric without cleaning up the asterisks, all those datapoints will be transformed to `NA` as it will be converting a character to a numeric. We will need to get rid of them first.
+
+Note that there are states that suppress proficiency rates for ethnicity/races whose populations are too small to report, denoted by special characters (‡). These will show up as `NA` when we change the column to numeric.
+
+- We 'mutate' the proficiency column by removing astericks in the string (`str_replace()`).
+- Now that the astericks are removed, we can transform the column to numeric using `as.numeric.` Dividing by 100 gives us a percentage.
 
 ```{r}
-head(naep17)
+naep17_tidy <-
+    naep17_tidy %>%
+    dplyr::mutate(at_or_above_proficient_clean =  stringr::str_replace(at_or_above_proficient, '\\*', '')) %>% # create a new column without *
```

Comment:
style-related and general. I like that you included the namespacing. wondering if we should do this across the board (perhaps it's something we add to our style guide #25)